### PR TITLE
DM-14429: Implement RFC-460: move afw.geom content to new geom package

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: examples/SConscript [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# -*- python -*-
+import os
+from lsst.sconsUtils import scripts, env, targets
+
+# This is more complicated than an examples SConscript should usually be (most can just use
+# BasicSConscript.examples() with no arguments), because we have a source file that's compiled
+# into both a C++ program and a SWIG module.
```
Comment (TallJimbo) [inline examples/SConscript]: Pre-existing, but `SWIG` -> `pybind11`.

File: include/lsst/geom.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/*
+ * An include file to include the header files for lsst::geom
+ */
```
Comment (TallJimbo) [inline include/lsst/geom.h]: Do we need this file for any kind of backwards compatibility?  If not, I think we should remove it; having these include-everything headers encourages downstream code to include more than it needs.

Comment (r-owen) [reply include/lsst/geom.h]: Many packages import afw/geom.h. I would expect the geom package to be small enough that this is a reasonable thing to do, but I can remove it if you feel very strongly about it.

Comment (TallJimbo) [reply include/lsst/geom.h]: I agree that it's less of a problem for this package than it is for others.  If other packages do use it, by all means leave it in.

**Comment by r-owen** [comment]
I will modernize the copyrights, add a sphinx doc build and fix imports.

Regarding whitespace issues in C++ files: I tried to run clang-format on all C++ code but I see I missed a few files and will fix those. Any remaining whiltespace issues are in my opinion not worth fixing; we should live with the output from clang-format.

**Comment by r-owen** [comment]
I cleaned up the issues you raised: add sphinx doc support including a preliminary index.rst, modernize copyright, standardize includes and ran clang-format on a few files I missed the first time. Rebased and ready to merge.

**Pull request merged**